packages/front-end/components/Settings/ConnectionSettings.tsx

```diff
@@ -142,48 +155,57 @@ export default function ConnectionSettings({
           projects={datasource?.projects || []}
         />
       );
-
+      break;
     case "snowflake":
-      return (
+      datasourceComponent = (
         <SnowflakeForm
           existing={existing}
           onParamChange={onParamChange}
           params={datasource?.params || {}}
         />
       );
-
+      break;
     case "clickhouse":
-      return (
+      datasourceComponent = (
         <ClickHouseForm
           existing={existing}
           onParamChange={onParamChange}
           setParams={setParams}
           params={datasource?.params || {}}
         />
       );
-
+      break;
     case "bigquery":
-      return (
+      datasourceComponent = (
         <BigQueryForm
           existing={existing}
           setParams={setParams}
           params={datasource?.params || {}}
           onParamChange={onParamChange}
         />
       );
-
+      break;
     case "mixpanel":
-      return (
+      datasourceComponent = (
         <MixpanelForm
           existing={existing}
           onParamChange={onParamChange}
           onManualParamChange={onManualParamChange}
           params={datasource?.params || {}}
         />
       );
-
+      break;
     default:
       invalidType = datasource.type;
       throw `Invalid type: ${invalidType}`;
   }
+  return (
+    <>
+      {datasourceComponent}
+      <SharedConnectionSettings
```

Comment:
I really like this solution, it won't trigger a re-test of the connection since it's affecting settings and not params.

However, you're adding it here to the "ConnectionSettings" which used to only affect those connection `params` which kicks off these test queries and stuff in the [putDataSource endpoint](https://github.com/growthbook/growthbook/blob/4b319072d263e77daef9bf9bd1eb17aba28cd359/packages/back-end/src/controllers/datasources.ts#L361).

There's 0 need to change your approach, but I feel like these could definitely live in `NewDataSourceForm` and `DataSourceForm`, where we already natively handle `settings` changes and keep them separate from `params` changes. Not really important at all, but wanted to call that out!

Reply:
I was looking at putting them there, but opted for this as it felt like the simplest option to avoid all of the conditional logic in the outer forms. I'm also hoping there are other shared settings we can use this for so it's not a one-off addition